docs/guides/quickfort-alias-guide.rst

```diff
@@ -0,0 +1,850 @@
+equivalently defined like this:
+
+::
+
+    nobins: C
+    nobarrels: E
+    nocontainers: {nobins}{nobarrels}
+
+Aliases used in alias definitions *must* be surrounded by curly brackets, even
+if they are the only text in the definition:
+
+::
+
+    alias1: text1
+    alias2: alias1
+    alias3: {alias1}
+
+Here, ``alias1`` and ``alias3`` expand to ``text1``, but ``alias2`` expands to
+the literal text ``alias1``.
+
+Keycodes
+~~~~~~~~
+
+Non-printable characters, like the arrow keys, are represented by their
+keycode name and are also surrounded by curly brackets, like ``{Right}`` or
+``{Enter}``. Keycodes are used exactly like aliases -- they just have special
+expansions that you wouldn't be able to write yourself. In order to avoid
+naming conflicts between aliases and keycodes, the convention is to start
+aliases with a lowercase letter.
+
+Any keycode name from the DF interface definition file
+(data/init/interface.txt) is valid, but only a few keycodes are actually
+useful for blueprints:
+
+::
+
+    Up
+    Down
+    Left
+    Right
+    Enter
+    ESC
+    Backspace
+    Space
+    Tab
+
+Repetitions
+~~~~~~~~~~~
+
+Anything enclosed within curly brackets can also have a number, indicating how
+many times that alias or keycode should be repeated. For example:
+``{togglesequence 9}`` or ``{Down 5}`` will repeat the ``togglesequence``
+alias nine times and the ``Down`` keycode five times, respectively.
+
+Modifier keys
+~~~~~~~~~~~~~
+
+Ctrl, Alt, and Shift modifiers can be specified for the next key by adding
+them into the key sequence. For example, Alt-h is written as ``{Alt}h``.
+
+Note that DF does not handle keys that have more than a single modifier, so
+sequences like ``{Ctrl}{Alt}a`` will result in an error.
+
+Shorthand characters
+~~~~~~~~~~~~~~~~~~~~
+
+Some frequently-used keycodes are assigned shorthand characters. Think of them
+as single-character aliases that don't need to be surrounded in curly
+brackets:
+
+::
+
+    &   expands to {Enter}
+    @   expands to {Shift}{Enter}
+    ~   expands to {Alt}
+    !   expands to {Ctrl}
+    ^   expands to {ESC}
+
+If you need literal versions of the shorthand characters, surround them in
+curly brackets, for example: use ``{!}`` for a literal exclamation point.
+
+Built-in aliases
+~~~~~~~~~~~~~~~~
+
+Most aliases that come with DFHack are in ``aliases-common.txt``, but there is
+one alias built into the code for the common shorthand for "make room":
+
+::
+
+    r+  expands to r+{Enter}
+
+This needs special code support since ``+`` can't normally be used in alias
+names. You can use it just like any other alias, either by itself in a cell
+(``r+``) or surrounded in curly brackets (``{r+}``).
+
+Sub-aliases
+~~~~~~~~~~~
+
+You can specify sub-aliases that will only be defined while the current alias
+is being resolved. This is useful for "injecting" custom behavior into the
+middle of a larger alias. For example:
+
+::
+
+    {quantumstopfromeast name="Trash Dump"}
```

Comment:
It wasn't initially clear to me from reading this section that the `quantumstopfromeast` alias would need to include `{name}` in its definition. Could help to provide an example (minimal) definition of this macro.

Also, what happens if the `name` parameter is omitted? In the `quantum_enable` example below, there is an existing alias of that name that serves to provide default behavior, but it's less clear to me what would happen in this case (I assume it would expand to an empty string?).

And finally, including underscores in the names of aliases here might make them a bit easier to read, but that's not hugely important (especially if you're trying to match aliases defined elsewhere).

Reply:
good feedback. text updated. I have it as a future task for myself to allow empty alias definitions so the "optional sub-alias" case is handled more elegantly.

underscores (and hyphens) are relatively new to alias names. all the "original" aliases are pure alphanumeric (since they used the same convention as Python Quickfort). I could rename the existing ones, but I would risk breaking blueprints that use them. I don't *think* there are many blueprints that would break at this point (there hasn't been very much time yet for users to create blueprints that use the alias library), but I can't be sure.